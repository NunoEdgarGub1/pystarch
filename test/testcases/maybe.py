def f(a):
    if a > 5:
        return 2 * a
    else:
        return None
def g(a):
    if a > 5:
        return None
    else:
        return 2 * a
def h(a):
    return g(a) if a > 5 else None
a = f(3)
b = g(3)
c = h(3)
