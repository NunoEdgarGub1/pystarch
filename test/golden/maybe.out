a NoneType None
b Num
c Maybe(Num)
f Function(a: Num -> Maybe(Union(Num,Str)))
g Function(a: Num -> Maybe(Union(Num,Str)))
h Function(a: Num -> Maybe(Union(Num,Str)))
